turbojpeg.py

```diff
@@ -283,6 +375,18 @@ def get_scaled_value(dim, num, denom):
             scaled_height = get_scaled_value(scaled_height, scaling_factor[0], scaling_factor[1])
         return scaled_width, scaled_height, jpeg_subsample, jpeg_colorspace
 
+    def __axis_to_image_boundaries(self, a, b, img_boundary, preserve, mcuBlock):
+        img_b = img_boundary - (img_boundary % mcuBlock)
+        if a > img_b:
+            a = img_b
+        else:
+            a = a - (a % mcuBlock)
+        if not preserve:
+            b = b + (a % mcuBlock)
```

Comment:
save away the original `a` or a copy of `a % mcuBlock`.  `a` and has already been moved to an mcu boundary by this point.